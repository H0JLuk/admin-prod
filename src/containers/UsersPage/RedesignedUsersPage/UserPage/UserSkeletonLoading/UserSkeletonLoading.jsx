import React from 'react';
import PropTypes from 'prop-types';
import { Skeleton } from 'antd';

import styles from './UserSkeletonLoading.module.css';

const UserSkeletonLoading = ({ showLastLabel , showCheckBoxes  }) => {
    const labelCount = showLastLabel ? 3 : 2;
    const checkBoxDiv = (
        <div className={ styles.rightSkeleton }>
            {Array.from({ length: 5 }, (_, i) => (
                <Skeleton.Input
                    key={ i }
                    style={ { width: 200, height: 25, marginTop: 25 } }
                    active
                />
            ))}
        </div>
    );

    return (
        <div className={ styles.skeleton }>
            <div>
                <div className={ styles.flexSkeleton }>
                    <Skeleton active paragraph={ { rows: 1 } } />
                    <div className={ styles.LabelsWrapperSkeleton }>
                        {Array.from({ length: labelCount }, (_, i) => (
                            <div className={ styles.labelsSkeleton } key={ i }>
                                <Skeleton.Input
                                    style={ { width: 200, height: 40 } }
                                    active
                                    size="small"
                                />
                                <Skeleton.Input
                                    style={ { width: 200, height: 40, marginLeft: 50 } }
                                    active
                                    size="small"
                                />
                            </div>
                        ))}
                    </div>
                </div>
                <div className={ styles.buttonsSkeleton }>
                    {Array.from({ length: 3 }, (_, i) => (
                        <Skeleton.Input
                            key={ i }
                            style={ { width: 190, height: 45, marginRight: 25, borderRadius: 50 } }
                            active
                        />
                    ))}
                </div>
            </div>
            {showCheckBoxes && checkBoxDiv}
        </div>
    );
};

UserSkeletonLoading.propTypes = {
    showLastLabel: PropTypes.bool,
    showCheckBoxes: PropTypes.bool,
};

UserSkeletonLoading.defaultProps = {
    showLastLabel: true,
    showCheckBoxes: true,
};

export default UserSkeletonLoading;
