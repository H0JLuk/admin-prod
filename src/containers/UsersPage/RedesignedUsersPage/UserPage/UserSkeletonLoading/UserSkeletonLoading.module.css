.skeleton {
    display: flex;
}

.flexSkeleton {
    display: flex;
    flex-direction: column;
}

.LabelsWrapperSkeleton {
    margin-top: 30px;
}

.labelsSkeleton {
    margin-top: 15px;
}

.buttonsSkeleton {
    margin-top: 40px;
}

.rightSkeleton {
    margin-top: 10px;
    display: flex;
    flex-direction: column;
}
