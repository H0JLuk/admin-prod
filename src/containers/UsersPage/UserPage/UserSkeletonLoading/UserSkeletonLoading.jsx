import React from 'react';
import { Skeleton } from 'antd';

import styles from './UserSkeletonLoading.module.css';

const titleStyle = { width: 400, height: 40 };
const inputsStyle = { height: 25 };
const blockTitleStyle = { height: 25 };
const checkboxStyle = { height: 25 };
const buttonsStyle = { height: 35 };
const secondWrapperStyle = { height: 220 };

const UserSkeletonLoading = () => (
    <div className={ styles.skeleton }>
        <Skeleton.Input style={ titleStyle } active size="small" />
        <div className={ styles.mainBlock }>
            <div className={ styles.labelsWrapperSkeleton }>
                { Array.from({ length: 3 }, (_, i) => (
                    <div className={ styles.labelsSkeleton } key={ i }>
                        <Skeleton.Input style={ inputsStyle } active size="small" />
                        <Skeleton.Input style={ inputsStyle } active size="small" />
                    </div>
                )) }
            </div>
            { /* <div className={ styles.labelsWrapperSkeleton }>
                { Array.from({ length: 1 }, (_, i) => (
                    <div className={ styles.labelsSkeleton } key={ i }>
                        <Skeleton.Input style={ inputsStyle } active size="small" />
                        <Skeleton.Input style={ inputsStyle } active size="small" />
                    </div>
                )) }
            </div> */ }
        </div>
        <div className={ styles.mainBlock }>
            <div className={ styles.labelsWrapperSkeleton } style={ secondWrapperStyle }>
                <div className={ styles.skeletonCheckBox }>
                    <div>
                        <Skeleton.Input style={ blockTitleStyle } active size="small" />
                    </div>
                    <div>
                        { Array.from({ length: 3 }, (_, i) => (
                            <Skeleton.Input key={ i } style={ checkboxStyle } active size="small" />
                        )) }
                    </div>
                    <div>
                        { Array.from({ length: 3 }, (_, i) => (
                            <Skeleton.Input key={ i } style={ checkboxStyle } active size="small" />
                        )) }
                    </div>
                    <div>
                        { Array.from({ length: 3 }, (_, i) => (
                            <Skeleton.Input key={ i } style={ checkboxStyle } active size="small" />
                        )) }
                    </div>
                </div>
            </div>
        </div>
        <div className={ styles.buttonsSkeleton }>
            { Array.from({ length: 3 }, (_, i) => (
                <Skeleton.Input key={ i } style={ buttonsStyle } active />
            )) }
        </div>
    </div>
);

export default UserSkeletonLoading;
