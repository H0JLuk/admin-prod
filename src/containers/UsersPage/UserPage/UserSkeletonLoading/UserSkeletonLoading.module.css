.skeleton {
    padding: 25px;
}

.mainBlock {
    margin-top: 30px;
    padding: 20px;
    background-color: #f7f7f7;
}

.labelsWrapperSkeleton {
    display: flex;
    width: 100%;
    height: 60px;
    max-width: 1100px;
}

.labelsWrapperSkeleton ~ .labelsWrapperSkeleton {
    margin-top: 16px;
}

.labelsSkeleton {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    flex-basis: 40%;
    margin-right: 10px;
}

.labelsSkeleton:first-child {
    flex-basis: 33%;
}

.labelsSkeleton :global(.ant-skeleton) {
    max-width: 300px;
    min-width: 240px;
}

.labelsSkeleton:first-child :global(.ant-skeleton) {
    min-width: 200px;
    max-width: 250px;
}

.skeletonCheckBox {
    display: flex;
    justify-content: space-between;
    width: 100%;
    max-width: 1100px;
    padding-top: 50px;
    padding-bottom: 30px;
}

.skeletonCheckBox > div {
    flex-basis: 25%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}

.skeletonCheckBox > div :global(.ant-skeleton-element) {
    max-width: 200px;
}

.buttonsSkeleton {
    margin-top: 25px;
}

.buttonsSkeleton > div {
    max-width: 150px;
    width: 100%;
}

.buttonsSkeleton > div:not(:last-child) {
    margin-right: 15px;
}
