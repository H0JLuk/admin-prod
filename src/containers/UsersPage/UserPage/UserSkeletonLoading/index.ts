import UserSkeletonLoading from './UserSkeletonLoading';

export default UserSkeletonLoading;
