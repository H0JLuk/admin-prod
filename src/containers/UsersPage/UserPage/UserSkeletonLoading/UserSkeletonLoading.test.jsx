import React from 'react';
import { shallow } from 'enzyme';
import UserSkeletonLoading from './UserSkeletonLoading';

describe('<UserSkeletonLoading /> test', () => {
    const Component = shallow(<UserSkeletonLoading />);

    it('should be mount snap', () => {
        expect(Component.html()).toMatchSnapshot();
    });
});
