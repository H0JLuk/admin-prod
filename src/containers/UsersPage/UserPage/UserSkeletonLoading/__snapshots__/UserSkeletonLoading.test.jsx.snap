// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserSkeletonLoading /> test should be mount snap 1`] = `"<div class=\\"skeleton\\"><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"width:400px;height:40px\\"></span></div><div class=\\"mainBlock\\"><div class=\\"labelsWrapperSkeleton\\"><div class=\\"labelsSkeleton\\"><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div></div><div class=\\"labelsSkeleton\\"><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div></div><div class=\\"labelsSkeleton\\"><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div></div></div><div class=\\"labelsWrapperSkeleton\\"><div class=\\"labelsSkeleton\\"><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div></div><div class=\\"labelsSkeleton\\"><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div></div><div class=\\"labelsSkeleton\\"><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div></div></div></div><div class=\\"mainBlock\\"><div class=\\"labelsWrapperSkeleton\\" style=\\"height:220px\\"><div class=\\"skeletonCheckBox\\"><div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div></div><div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div></div><div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div></div><div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input ant-skeleton-input-sm\\" style=\\"height:25px\\"></span></div></div></div></div></div><div class=\\"buttonsSkeleton\\"><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input\\" style=\\"height:35px\\"></span></div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input\\" style=\\"height:35px\\"></span></div><div class=\\"ant-skeleton ant-skeleton-element ant-skeleton-active\\"><span class=\\"ant-skeleton-input\\" style=\\"height:35px\\"></span></div></div></div>"`;
